include/zephyr/usb/usbh.h

```diff
@@ -137,6 +137,24 @@ int usbh_disable(struct usbh_contex *uhs_ctx);
  */
 int usbh_shutdown(struct usbh_contex *const uhs_ctx);
 
+/**
+ * @brief Helper function to get device's address
+ *
+ * @param[in] usb_device Pointer to USB device
+ *
+ * @return the device's address.
+ */
+uint8_t usbh_udev_get_address(void *usb_device);
```

Comment:
Why `void *usb_device`? Shouldn't it simply be `struct usb_device *usb_device`?

Reply:
It is called from context that can't see `struct usb_device`. For example: the uhc files that can't see `struct usb_device`.

Reply:
Just treat it as opaque pointer then. Having `void *` here is bad.

Reply:
In the `struct uhc_transfer`, it is `void *udev`. So I use the `void *`. Could you please give reference codes how to treat it as opaque pointer?

Reply:
That would need a change in the `struct uhc_transfer` then. Opaque pointers in C are simply achieved by having incomplete struct definition. So in this case it would be `struct usb_device;` above `struct uhc_transfer {` and then the `void *udev;` can be changed to `struct usb_device *udev;`. Similar could be done with the transfer completion callback. I somehow completely missed this during the uhc review.

Reply:
Thanks @tmon-nordic, I changed it, please help to review again.